compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/kotlin/KSTypeImpl.kt

```diff
@@ -77,7 +81,18 @@ class KSTypeImpl private constructor(
     override fun isAssignableFrom(that: KSType): Boolean {
         val subType = (that as? KSTypeImpl)?.kotlinType ?: return false
         ResolverImpl.instance.incrementalContext.recordLookupWithSupertypes(subType)
-        return subType.isSubtypeOf(kotlinType)
+        val declarationDescriptor = kotlinType.constructor.declarationDescriptor
+        // Compiler subtype checking does not convert Java types to Kotlin types, while getting super types
+        // from a java type does the conversion, therefore resulting in subtype checking for Java types to fail.
+        // Check if candidate super type is a Java type, convert to Kotlin type for subtype checking.
+        val thisType = if (declarationDescriptor is JavaClassDescriptor) {
+            JavaToKotlinClassMapper
+                .mapJavaToKotlin(declarationDescriptor.fqNameSafe, ResolverImpl.instance.module.builtIns)
+                ?.defaultType
+                ?.replace(kotlinType.arguments)
```

Comment:
Does this need to be done recursively, e.g., is `java.util.Container<java.util.Set>` assignable from `kotlin.collections.Set<kotlin.collections.Set>`?

Reply:
I thought about this case when writing test, then realized it is impossible to do so, because we are only getting Java types via `getJavaClassByName`, from where we can only get a `KSType`, and because we only support build type argument with a type reference, therefore KSType can't be used to build a type argument. Which in turn, means that I can only try to get such test case in code, and while resolving types in source, all these types in the type arguments are already converted to kotlin types, therefore it is impossible to reproduce this in test case, nor in real practice.

Reply:
Except that we have `Resolver.createKSTypeReferenceFromKSType()` 😢